src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -51,6 +51,15 @@
  * components are classified as different kind of variables by
  * <a href="https://docs.oracle.com/javase/specs/jls/se17/html/index.html">JLS</a>.
  * </p>
+ * <ul>
+ * <li>
+ * Property {@code allowUnnamedVariables} - Allow
+ * <a href="https://docs.oracle.com/en/java/javase/21/docs/specs/unnamed-jls.html">
+ * unnamed variables</a> to not be violated as unused.
```

Comment:
> "Allow unnamed variables to be unused.".

sounds weird to me. because unnamed variables can't be used by definition, this is something we can't allow or disallow. 
what we allow or disallow is treating it as a violation or not

Reply:
> unnamed variables can't be used by definition

I was thinking more along the lines of also old Java versions where `_` could be assigned. Do we need to be more specific and says "single underscore variables"? @nrmancuso What you think?

Reply:
yes IMO if we want to be accurate on this we should use "underscore" not "unnamed"

Reply:
`AllowUnnamedVariables` - do not place violation on variables named with a single underscore (known as `unnamed variables in Java 21+).

Reply:
done

Reply:
@nrmancuso I am un-resolving this. Property documentation need to start with a verb, otherwise the generated javadocs don't make sense.

> Do not place violations on variables ...
> Setter to do not place violations ...

Reply:
> @nrmancuso I am un-resolving this. Property documentation need to start with a verb, otherwise the generated javadocs don't make sense.
> 
> > Do not place violations on variables ...
> > Setter to do not place violations ...

Ah shoot. Yes, this is nonsense. 

`Allow variables named with a single underscore (known as unnamed variables in Java 21+). `

Reply:
done

Reply:
Looks good.